script/grblcon.py

```diff
@@ -0,0 +1,527 @@
+        self._status_display.set_text(('status', statstring))
+
+    def _process_realtime_state(self, line):
+        """Handles realtime state updates
+
+        :param line: a string containing a state message
+        :returns: None
+        :rtype: None
+
+        """
+        # rip out the tag delmiters
+        line = line.strip('<>')
+
+        # Split the line into discreet sections
+        (self._status.state,
+         m_pos,
+         w_pos,
+         self._status.line) = line.split(':')
+
+        self._status.mach_positions = Positions(m_pos)
+        self._status.work_positions = Positions(w_pos)
+
+    def _process_limit_flags(self, line):
+        """Handles incoming limit flag information
+
+        :param line: a string containing the limit flag message
+        :returns: None
+        :rtype: None
+
+        """
+        self._status.limit_flags = line.strip('/')
+
+    def _check_and_clean_line(self, line):
+        """Performs a checksum on serial line data and filters bad responses
+
+        :param line: A string in the format [data0..n][checksum][\n]
+        :returns: None if bad checksum, else stripped string
+        :rtype: None/str
+
+        """
+        checksum, _ = line[-2:]
+        line = line[:-2]
+
+        calcsum = sum([ord(x) for x in line]) & 0xff
+        residue = calcsum ^ ord(checksum)
+
+        if residue:
+            err = 'Bad Checksum. {} != {}. Data: {}'.format(
+                calcsum,
+                ord(checksum),
+                line.strip()
+            )
+            self._log_error(err)
+            return None
+
+        return line.strip()
+
+    def _process_serial_line(self, bytestring):
+        """Sorts and handles lines of data from the grbl firmware
+
+        :param bytes: a bytearray in the form of a newline terminated string
+        :returns: None
+        :rtype: None
+
+        """
+
+        line = self._check_and_clean_line(bytestring.decode('latin'))
+
+        # Short circuit in case we got a bad checksum
+        if not line:
+            return
+
+        # Assume that we'll filter the message, we'll use this later
+        # as a part of the squelch check
+        filtered = True
+
+        if line.startswith('<'):   # Real time state
+            self._process_realtime_state(line)
+        elif line.startswith('/'):  # Limit pin state
+            self._process_limit_flags(line)
+        else:
+            filtered = False
+            self._update_text_display(line)
+
+        if filtered and not self._quiet:
+            self._update_text_display(line)
+
+        self._update_status()
+
+    def _serial_rx(self):
+        if not self._port:
+            return
+
+        self._write_fd = self.loop.watch_pipe(self._process_serial_line)
+        while not self._end_evt.isSet():
+            line = self._port.readline()
+            if self._write_fd and line:
+                os.write(self._write_fd, bytes(line))
+
+    def _open_port(self, portstr, baud):
```

Comment:
Might be useful to make some default arguments of the standard serial port and buad rate.  Not sure if this is the best place for that, but I still want to bring it up.